# TCSS 478/588 In-class Exercise: Logistic regression

These exercises will be used to illustrate how to perform data analyses using R.
Alternatively, you can use R or Rstudio. 
I assume you will learn basic R commands on your own.

Credit: The data and ideas behind these exercises and homeworks are from the NIH LINCS DCIC Crowdsourcing Portal and Ma’ayan Lab @ Mt Sinai, New York.
http://www.maayanlab.net/crowdsourcing/megatask1.php

The overarching goal is to predict adverse drug reactions. But we will take this task step by step. You will learn more about the meanings of these data in future lectures. For illustrative purposes in today’s class, just treat these data as numbers in a matrix.

## 1. Download and read in gene expression data (same as last week)
Download the data “gene_expression_n438x978.txt” from canvas to your computer.  I have removed the first character (#) and an ‘ from the input file to make R happy.

In [1]:
getwd()

[1] "/Users/kayee/UW-Tacoma/TCSS478/kayee-winter2019/notebooks/ADR"

In [2]:
my.data <- read.table ("gene_expression_n438x978.txt", sep="\t", header=T)
dimnames(my.data)[[1]] <- my.data[,1]
my.data <- my.data[, -1]
my.data[1:5, 1:6]

PSME1        ATF1         RHEB         FOXO3      
ACETAZOLAMIDE       -0.015159099 -0.031470528 -0.004733488  0.02591061
IRBESARTAN          -0.026811981  0.012151979 -0.025550148 -0.02401181
IPRATROPIUM BROMIDE  0.001017958 -0.008650622 -0.018128698 -0.02079971
EFAVIRENZ           -0.004398264  0.055387992  0.004658520  0.01380732
THIAMINE             0.001838965 -0.018079188 -0.011855532 -0.03705033
                    RHOA          IL1B        
ACETAZOLAMIDE        0.0056296773 -0.007325439
IRBESARTAN          -0.0106717396  0.070951154
IPRATROPIUM BROMIDE -0.0002722781  0.025003360
EFAVIRENZ           -0.0340697348  0.024005885
THIAMINE            -0.0133954959 -0.001752450

In [3]:
dim (my.data)

[1] 438 978

# 2. Download and read in the adverse drug reaction (0/1) data
Download the dataset "ADRs_HLGT_n438x232.txt" from Canvas to your computer.

In [4]:
adr.mat <- read.table ("ADRs_HLGT_n438x232.txt", sep="\t", header=T)
dimnames(adr.mat)[[1]] <- adr.mat [, 1]
adr.mat <- adr.mat[, -1]
dim (adr.mat)
adr.mat[1:5, 1:6]

[1] 438 232

Abdominal.hernias.and.other.abdominal.wall.conditions
ACETAZOLAMIDE       0                                                    
IRBESARTAN          0                                                    
IPRATROPIUM BROMIDE 0                                                    
EFAVIRENZ           0                                                    
THIAMINE            0                                                    
                    Abortions.and.stillbirth Acid.base.disorders
ACETAZOLAMIDE       0                        1                  
IRBESARTAN          0                        0                  
IPRATROPIUM BROMIDE 0                        0                  
EFAVIRENZ           0                        0                  
THIAMINE            0                        0                  
                    Administration.site.reactions Adrenal.gland.disorders
ACETAZOLAMIDE       1                             0                      
IRBESARTAN          0                             1                      
IPRATROPIUM BROMIDE 0                             1                      
EFAVIRENZ           0                             1                      
THIAMINE            0                             0                      
                    Age.related.factors
ACETAZOLAMIDE       0                  
IRBESARTAN          0                  
IPRATROPIUM BROMIDE 0                  
EFAVIRENZ           1                  
THIAMINE            0

## 3. Your task: how would you explore this adr.mat matrix? This is a binary matrix. How sparse is this matrix?

## 4. Exercise (for home): Exercise: are there any side effects that are not associated with any of the 438 drugs? What is the average number of drugs associated with each side effects?

# 5. Logistic regression

## Is CDK1 predictive of heart failures?